lib/rbac/filterer.rb

```diff
@@ -416,6 +416,7 @@ def get_belongsto_filter_object_ids(klass, filter)
     end
 
     def get_managed_filter_object_ids(scope, filter)
+      return scope.where(filter.to_sql.first) if filter.kind_of?(MiqExpression)
```

Comment:
@lpichler, yes, these expressions will be limited to tags on the main class only and will always be convertible to SQL.